core/modules/catalog/helpers/search.ts

```diff
@@ -0,0 +1,129 @@
+import Vue from 'vue'
+import { Logger } from '@vue-storefront/core/lib/logger';
+import config from 'config'
+import omit from 'lodash-es/omit'
+import { configureProductAsync, calculateTaxes } from '@vue-storefront/core/modules/catalog/helpers'
+import { quickSearchByQuery } from '@vue-storefront/core/lib/search'
+import { StorageManager } from '@vue-storefront/core/store/lib/storage-manager'
+import { entityKeyName } from '@vue-storefront/core/store/lib/entities'
+import SearchQuery from '@vue-storefront/core/lib/search/searchQuery'
+
+const canCache = ({ includeFields, excludeFields }) => {
+  const isCacheable = (includeFields === null && excludeFields === null)
+
+  if (isCacheable) {
+    Logger.debug('Entity cache is enabled for productList')()
+  } else {
+    Logger.debug('Entity cache is disabled for productList')()
+  }
+
+  return isCacheable
+}
+
+const getCacheKey = (product, cacheByKey) => {
+  if (!product[cacheByKey]) {
+    cacheByKey = 'id'
+  }
+
+  return entityKeyName(cacheByKey, product[(cacheByKey === 'sku' && product['parentSku']) ? 'parentSku' : cacheByKey]) // to avoid caching products by configurable_children.sku
+}
+
+const configureChildren = (product) => {
+  if (product.configurable_children) {
+    for (let configurableChild of product.configurable_children) {
+      if (configurableChild.custom_attributes) {
+        for (let opt of configurableChild.custom_attributes) {
+          configurableChild[opt.attribute_code] = opt.value
+        }
+      }
+    }
+  }
+
+  return product
+}
+
+const configureTaxes = (context, { resp, isCacheable, cacheByKey }) => {
```

Comment:
basically if helper needs a vuex context, then it should be an action not helper. Otherwise helpers instead of helping they are distracting vuex logic to parts.
Helpers should do computations etc. - take data and return result, which is also easy to test by unit tests.
But vuex actions should be the ones, which are dispatching other actions.
Let me know if you understand this point of view - it is really important to have this understanding of helpers.
cc @pkarw @filrak @przspa @lukeromanowicz

Reply:
@patzick completely agree, basically I wanted to ask about this point just to clarify what helpers means in this project. My understanding was exactly like yours but I saw context-bound in the other helpers so I thought that sometimes you are using it in that way and that's allowed. Ok I'll try to refactor it as much as it is possible, because probably some of them would be useful later.

Reply:
yes, we have that kind of helpers, but we're trying to refactor them in time with this conception. That's cool that you see this the same way. I agree that some of them are reusable, but only with logic part - with every dispatch there should be an action, which can also be reusable if is doing one specific thing at the time not set of things :)